src/System.Management.Automation/security/Authenticode.cs

```
@@ -437,89 +443,81 @@ private static Signature GetSignatureFromWinVerifyTrust(string fileName, string
-                Marshal.StructureToPtr(wtd, wtdBuffer, false);
-
-                // The result is returned to the caller, and handled generically.
-                // Disable the PreFast check for Win32 error codes, as we don't care.
-#pragma warning disable 56523
-                dwResult = NativeMethods.WinVerifyTrust(
-                    IntPtr.Zero,
-                    WINTRUST_ACTION_GENERIC_VERIFY_V2,
-                    wtdBuffer);
-#pragma warning restore 56523
-
-                wtData = Marshal.PtrToStructure<NativeMethods.WINTRUST_DATA>(wtdBuffer);
-            }
-            finally
-            {
-                Marshal.DestroyStructure<Guid>(WINTRUST_ACTION_GENERIC_VERIFY_V2);
-                Marshal.FreeCoTaskMem(WINTRUST_ACTION_GENERIC_VERIFY_V2);
-                Marshal.DestroyStructure<NativeMethods.WINTRUST_DATA>(wtdBuffer);
-                Marshal.FreeCoTaskMem(wtdBuffer);
             }
-
-            return dwResult;
         }
 
         [ArchitectureSensitive]
         private static X509Certificate2 GetCertFromChain(IntPtr pSigner)
         {
-            X509Certificate2 signerCert = null;
-
-            // We don't care about the Win32 error code here, so disable
-            // the PreFast complaint that we're not retrieving it.
-#pragma warning disable 56523
-            IntPtr pCert =
-                NativeMethods.WTHelperGetProvCertFromChain(pSigner, 0);
-#pragma warning restore 56523
-
-            if (pCert != IntPtr.Zero)
+            try
             {
+                IntPtr pCert = WinTrustMethods.WTHelperGetProvCertFromChain(pSigner, 0);
                 NativeMethods.CRYPT_PROVIDER_CERT provCert =
                     Marshal.PtrToStructure<NativeMethods.CRYPT_PROVIDER_CERT>(pCert);
-                signerCert = new X509Certificate2(provCert.pCert);
+                return new X509Certificate2(provCert.pCert);
+            }
+            catch (Win32Exception)
+            {
+                return null;
```

Please add a comment in code what errors we ignore and why.

---

I'm not really sure why the error was ignored but it is today so I just essentially replicated here.

https://github.com/PowerShell/PowerShell/blob/8d453da283162a32f0296b49068c247b2d41b315/src/System.Management.Automation/security/Authenticode.cs#L497-L516

I've added a similar comment to the new function though.